platforms/nuttx/src/px4/common/SerialImpl.cpp

```diff
@@ -339,7 +339,7 @@ ssize_t SerialImpl::write(const void *buffer, size_t buffer_size)
 	}
 
 	int written = ::write(_serial_fd, buffer, buffer_size);
-	::fsync(_serial_fd);
+	tcdrain(_serial_fd); // Wait until all output is transmitted
```

Comment:
This looks fine. The original fsync was copied from the GPS serial driver I believe. But this change will also need to be done on the Posix SerialImpl.cpp file as well.